executor.py

```diff
@@ -82,8 +84,22 @@ def search(
         :param parameters: Dictionary to define the `filter` that you want to use.
         :param kwargs: additional kwargs for the endpoint
         """
-        limit = int(parameters.get('limit', self.limit))
-        docs.match(self._index, filter=parameters.get('filter', None), limit=limit)
+        match_args = (
+            {**self._match_args, **parameters}
+            if parameters is not None
+            else self._match_args
+        )
+        match_args = ElasticSearchIndexer._filter_match_params(docs, match_args)
+        docs.match(self._index, filter=parameters.get('filter', None), **match_args)
+
+
+    @staticmethod
+    def _filter_match_params(docs, match_args):
+        # get only those arguments that exist in .match
+        args = set(inspect.getfullargspec(docs.match).args)
```

Comment:
I do not think this is needed, specially if `match` can get different `kwargs` I would just pass it.

Also, if this is necessary, this `args` can be something precomputed, do  not need to do this expensive inspection every time